pulp_rpm/app/tasks/publishing.py

```diff
@@ -95,21 +92,46 @@ def prepare_metadata_files(self, contents, folder=None):
 
         return repomdrecords
 
-    def get_packages(self, contents):
+    def publish_artifacts(self, content):
         """
-        Get packages from content.
+        Publish artifacts.
 
         Args:
-            contents (pulpcore.plugin.models.Content): A list of contents.
+            content (pulpcore.plugin.models.Content): content set.
 
-        Returns:
-            packages (pulp_rpm.models.Package): A list of packages.
+        """
+        published_artifacts = []
+        for content_artifact in ContentArtifact.objects.filter(content__in=content).iterator():
+            published_artifacts.append(PublishedArtifact(
+                relative_path=content_artifact.relative_path,
+                publication=self.publication,
+                content_artifact=content_artifact)
+            )
 
+        PublishedArtifact.objects.bulk_create(published_artifacts, batch_size=2000)
+
+    def handle_sub_repo(self, lst, identifier):
         """
-        packages = Package.objects.filter(pk__in=contents).\
-            prefetch_related('contentartifact_set')
+        Handle sub repos.
+
+        Get content and publish them.
+
+        Args:
+            lst (pulp_rpm.models.Addon/Variant): A list of addons/variants.
+            identifier (str): field name of model id
 
-        return packages
+        """
+        sub_repos_pks = []
+        for item in lst:
```

Comment:
same here, can we make it more descritpive, e.g. `item` -> `sub_repo`?

Reply:
item would be addon/variant, I don't know which generic name could cover it